contracts/integrations/custom/CustomIntegrationBalancerv2.sol

```diff
@@ -0,0 +1,416 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity 0.7.6;
+pragma experimental ABIEncoderV2;
+
+import {IAsset} from '@balancer-labs/v2-vault/contracts/interfaces/IAsset.sol';
+import {BasePool} from '@balancer-labs/v2-pool-utils/contracts/BasePool.sol';
+import {IERC20} from '@balancer-labs/v2-solidity-utils/contracts/openzeppelin/IERC20.sol';
+import {ERC20} from '@balancer-labs/v2-solidity-utils/contracts/openzeppelin/ERC20.sol';
+import {IVault} from '@balancer-labs/v2-vault/contracts/interfaces/IVault.sol';
+import {WeightedMath} from '@balancer-labs/v2-pool-weighted/contracts/WeightedMath.sol';
+
+import {IBabController} from '../../interfaces/IBabController.sol';
+import {CustomIntegration} from './CustomIntegration.sol';
+import {PreciseUnitMath} from '../../lib/PreciseUnitMath.sol';
+import {LowGasSafeMath} from '../../lib/LowGasSafeMath.sol';
+import {BytesLib} from '../../lib/BytesLib.sol';
+import {ControllerLib} from '../../lib/ControllerLib.sol';
+
+/**
+ * @title Custom integration for the Balancer V2 protocol
+ * @author ChrisiPK, MartelAxe
+ *
+ * This integration allows Babylon Finance gardens to provide liquidity to Balancer V2 pools.
+ */
+contract CustomIntegrationBalancerv2 is CustomIntegration, WeightedMath {
+    using LowGasSafeMath for uint256;
+
+    /* ============ State Variables ============ */
+
+    address private constant vaultAddress = 0xBA12222222228d8Ba445958a75a0704d566BF2C8;
+
+    // Allowable slippage for the price of pool tokens in percent.
+    uint8 private constant priceSlippage = 5;
```

Comment:
We have defined the slippage for the result token price and the minimum output amounts here. Is there some existing value that we should be using instead? We saw that there is a constant `SLIPPAGE_ALLOWED` in `CustomIntegration`, but seems that is not used anywhere?

Reply:
Yeah, I would use that constant and in 1e18 precision

Reply:
Okay, changed to use that constant.